tests/datasource/test_new_datasource_with_runtime_data_connector.py

```diff
@@ -0,0 +1,499 @@
+            Batch
+        ] = basic_datasource_with_runtime_data_connector.get_batch_list_from_batch_request(
+            batch_request=batch_request
+        )
+
+
+def test_set_data_asset_name_for_runtime_data(
+    basic_datasource_with_runtime_data_connector,
+):
+    test_df: pd.DataFrame = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    partition_request: dict
+    partition_request = {
+        "partition_identifiers": {
+            "pipeline_stage_name": "core_processing",
+            "airflow_run_id": 1234567890,
+            "custom_key_0": "custom_value_0",
+        }
+    }
+
+    # set : my_runtime_data_asset
+    batch_request: dict = {
+        "datasource_name": basic_datasource_with_runtime_data_connector.name,
+        "data_connector_name": "test_runtime_data_connector",
+        "data_asset_name": "my_runtime_data_asset",
+        "batch_data": test_df,
+        "partition_request": partition_request,
+        "limit": None,
+    }
+    batch_request: BatchRequest = BatchRequest(**batch_request)
+    batch_list: List[
+        Batch
+    ] = basic_datasource_with_runtime_data_connector.get_batch_list_from_batch_request(
+        batch_request=batch_request
+    )
+    assert batch_list[0].batch_definition.data_asset_name == "my_runtime_data_asset"
+
+
+def test_get_available_data_asset_names(basic_datasource_with_runtime_data_connector):
+    expected_available_data_asset_names: dict[List[str]] = {
+        "test_runtime_data_connector": ["IN_MEMORY_DATA_ASSET"]
+    }
+    available_data_asset_names: List[
+        dict
+    ] = basic_datasource_with_runtime_data_connector.get_available_data_asset_names()
+    assert available_data_asset_names == expected_available_data_asset_names
+
+
+def test_get_batch_definition_list_from_batch_request_length_one(
+    basic_datasource_with_runtime_data_connector,
+):
+    test_df: pd.DataFrame = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+
+    partition_request: dict = {
+        "partition_identifiers": {
+            "airflow_run_id": 1234567890,
+        }
+    }
+
+    batch_request: dict = {
+        "datasource_name": basic_datasource_with_runtime_data_connector.name,
+        "data_connector_name": "test_runtime_data_connector",
+        "data_asset_name": "IN_MEMORY_DATA_ASSET",
+        "batch_data": test_df,
+        "partition_request": partition_request,
+        "limit": None,
+    }
+    batch_request: BatchRequest = BatchRequest(**batch_request)
+    batch_list: List[
+        Batch
+    ] = basic_datasource_with_runtime_data_connector.get_batch_list_from_batch_request(
+        batch_request=batch_request
+    )
+    # batches are a little bit more difficult to test because of batch_markers
+    # they are ones that uniquely identify the data
+    assert len(batch_list) == 1
+
+
+def test_get_batch_with_pipeline_style_batch_request_interrogate_data_connector_more_than_once(
+    basic_datasource_with_runtime_data_connector,
+):
```

Comment:
I do not think that this test probes the presence of data_asset_name sufficiently deeply. While it may exist in a BatchDefinition, is it attributed to the given Batch/Partition and can it be queried from it across multiple calls to the given data connector? I feel that needing more time to figure this out properly was one of the main reasons why we left this unfinished during BatchPod (and had to settle for IN_MEMORY_DATA_ASSET as the default).

Hi @alexsherstinsky this is a first attempt at addressing your comment ^^^^ 

Could you have a look? I'd be happy to discuss further